scripts/delete.coffee

```diff
@@ -12,9 +20,14 @@ class Client
 
   loadRecentMessages: (count) ->
     count = count || 50
-    url = "https://slack.com/api/channels.history?" +
-      "token=#{@token}&channel=#{@channel}&count=#{count}"
     deferred = Q.defer()
+    endpoint = switch @channel[0]
+      when 'C' then 'channels.history'
+      when 'G' then 'groups.history'
+      when 'D' then 'im.history'
+      else deferred.reject "Unknown channel type for channel #{@channel}"
+    url = "https://slack.com/api/#{endpoint}?" +
```

Comment:
This is flawed in that it will send a bad request to slack if the channel type is unknown. Could add some exception handling or other control flow to make it so the request is not issued.